modeldb/modeldb.py

```diff
@@ -31,7 +33,41 @@ def download_model(model_id):
         )
         with open(model_zip_uri, "wb+") as zipfile:
             zipfile.write(base64.standard_b64decode(url["file_content"]))
-    except Exception as e:   #  noqa
+
+        if "github" in model_run_info:
+            # This means we should try to replace the version of the model that
+            # we downloaded from the ModelDB API just above with a version from
+            # GitHub
+            github = model_run_info["github"]
+            if github == "default":
+                suffix = ""
+            elif github.startswith("pull/"):
+                pr_number = int(github[5:])
+                suffix = "/pull/{}/head".format(pr_number)
+            else:
+                raise Exception("Invalid value for github key: {}".format(github))
+            github_url = "https://api.github.com/repos/ModelDBRepository/{model_id}/zipball{suffix}".format(
+                model_id=model_id, suffix=suffix
+            )
+            # Replace the local file `model_zip_uri` with the zip file we
+            # downloaded from `github_url`
+            num_attempts = 3
+            status_codes = []
```

Comment:
It happened a few times (a week or two ago, I haven't been trying in the last few days).

As well as trying to increase the chances of success, this was also trying to make failures more obvious -- the scripts catch exceptions when fetching models, so it can unfortunately be a bit non-obvious when something does go wrong...